common/scpi_cmds.c

```diff
@@ -61,6 +64,70 @@ static uint32_t scpi_cmd_get_scp_cap_tx_payload[] = {
 	0,
 };
 
+/*
+ * Handler/payload data for SCPI_CMD_SET_CSS_PWR: Set CSS power state.
+ *
+ * The power state provided by PSCI is already coordinated. Simply turn things
+ * on from highest to lowest power level, then turn things off from lowest to
+ * highest power level. This ensures no power domain is turned on before its
+ * parent, and no power domain is turned off before any of its children.
+ */
+uint32_t
+scpi_cmd_set_css_pwr_handler(uint32_t *rx_payload, uint16_t rx_size __unused,
+                             uint32_t *tx_payload __unused,
+                             uint16_t *tx_size __unused)
+{
+	uint32_t descriptor    = rx_payload[0];
+	uint8_t  core          = (descriptor >> 0x00) & BITMASK(0, 4);
+	uint8_t  cluster       = (descriptor >> 0x04) & BITMASK(0, 4);
+	uint8_t  core_state    = (descriptor >> 0x08) & BITMASK(0, 4);
+	uint8_t  cluster_state = (descriptor >> 0x0c) & BITMASK(0, 4);
+	uint8_t  css_state     = (descriptor >> 0x10) & BITMASK(0, 4);
+
+	/* Do not check if the CSS should be turned on, as receiving this
```

Comment:
Detailed description comments should follow the [Doxygen specification](https://www.stack.nl/~dimitri/doxygen/manual/docblocks.html) (separate from but similar to in-body descriptions).

Reply:
Doxygen is for API documentation. This isn't a description of the function, but an optimization implementation detail.

Reply:
> Doxygen is for API documentation.

Where are you getting this from? Doxygen can be used for _only_ API documentation _or_ it can generate docs for _all_ source code documentation.

Reply:
Yes, but this isn't intended to go in the documentation. It's a comment about something tricky I did, that didn't happen to fit on one line.

Reply:
Whether or not this code is included in the generated documentation (really, this is still yet to be decided), it's the style we have established to use since #94.

Reply:
Which says:

> Public macros, function prototypes, and declarations should be documented according to Doxygen JavaDoc-style comment blocks.

**Doxygen is for API**. **This is not API**. This is not a "detailed description comment". This is none of those things! Do you really not understand the difference between "useful information for people who use this function" and "justification for how this function was written"? Do you really want me to change every single one-line comment in the code into a "comment block"? It's only two lines long because it doesn't fit on one line.

This is a stupid thing to waste time over.

Reply:
> Doxygen is for API. This is not API.

Again Doxygen is not _only_ for API.

> detailed description comment

You're right - this is an in-body comment (which Doxygen has outlined a style for).

> Do you really want me to change every single one-line comment in the code into a "comment block"?

If you made the change, we wouldn't still be talking about this. This is a matter that should be discussed offline. To keep this brief, if you request changes on other PRs that address comment style, this is a valid change. 

I am also fine with not blocking PRs over comment styles, but all org members need to be on the same page about this.